backend/fms_core/viewsets/sample.py

```diff
@@ -324,14 +324,38 @@ def list_export(self, _request):
         serialized_data = self.fetch_export_data()
         return Response(serialized_data)
 
+    @action(detail=False, methods=["get"])
+    def list_export_metadata(self, _request):
+        self.queryset = self.filter_queryset(self.get_queryset())
+        serialized_data = self.fetch_export_metadata()
+        return Response(serialized_data)
+
     def get_renderer_context(self):
         context = super().get_renderer_context()
         if self.action == 'list_export':
             fields = SampleExportSerializer.Meta.fields
             context['header'] = fields
             context['labels'] = {i: i.replace('_', ' ').capitalize() for i in fields}
+        elif self.action == 'list_export_metadata':
+            fields = self.get_metadata_fields()
+            context['header'] = fields
+            context['labels'] = {i: i.replace('_', ' ').capitalize() for i in fields}
         return context
 
+    def get_metadata_fields(self):
```

Comment:
This is a likely source of bugs ... we should store the columns after returning from the fetch into a self variable and pick it from there in the get_renderer_context ... otherwise we risk having synchronization issues between fetch and this function.